File: consume/hexsplit.py
#!/usr/bin/python3

import os
import sys

# Parsing the variable

def _dequote(part):
    subs = part.split("%")
    # Will raise if there are <2 chars after % or if these aren't valid hex
    return subs[0] + "".join(chr(int(sub[0:2], 16)) + sub[2:] for sub in subs[1:])

def decode(prefix_str):
    tuples = (part.split("=") for part in prefix_str.split(":") if part) if prefix_str else ()
    # Will raise if any tuple can't be destructured into a pair
    return [(_dequote(src), _dequote(dst)) for src, dst in tuples]

pm = decode(os.getenv("BUILD_PATH_PREFIX_MAP", ""))

# Applying the variable

def map_prefix(string, pm):
    for src, dst in reversed(pm):
        if string.startswith(src):
            return dst + string[len(src):]
    return string

for v in sys.argv[1:]:
    print(map_prefix(v, pm))

File: consume/hexsplit.js
#!/usr/bin/nodejs

var unquote = function(x) {
    if (x.indexOf("%00") >= 0) throw "invalid value: " + x;
    var parts = x.split(/=/g).map(decodeURIComponent);
    if (parts.length !== 2) throw "invalid value: " + x;
    return parts;
};

var parse_prefix_map = function(x) {
    return x ? x.split(/:+/g).map(unquote) : [];
}

var map_prefix = function(string, pm) {
    for (var i = pm.length - 1; i >= 0; --i) {;
        var src = pm[i][0];
        var dst = pm[i][1];
        if (string.indexOf(src) === 0) {
            return dst + string.substr(src.length);
        }
    }
    return string;
}

var pm = parse_prefix_map(process.env["BUILD_PATH_PREFIX_MAP"]);

// var i = 2 is just how nodejs yolos its way through common conventions
for (var i = 2, l = process.argv.length; i < l; ++i) {
    console.log(map_prefix(process.argv[i], pm));
}
